### PR TITLE
Hdrp stacklit custom specular occlusion

### PR DESCRIPTION
**Pull request by slunity**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

StackLit: Adding custom user calculated specular occlusion input port (one value for all lobes for now)
Set default baked specular occlusion to none, like for Lit.

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?
Ran local tests only. Katana in progress.

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=hdrp-stacklit-custom-specular-occlusion&automation-tools_branch=master&unity_branch=trunk&sort=1-asc

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
None

**Halo Effect**: None, Low, Medium, High?
Low
---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/StackLitMasterNode.cs [hunk 1]
```diff
@@ -172,13 +172,19 @@ public enum SpecularOcclusionBaseMode
             DirectFromAO = 1, // TriACE
             ConeConeFromBentAO = 2,
             SPTDIntegrationOfBentAO = 3,
+            Custom = 4,
+            // Custom user port input: For now, we will only have one input used for all lobes and only for data-based SO
+            // (TODO: Normally would need a custom input per lobe.
+            // Main rationale is that roughness can change IBL fetch direction and not only BSDF lobe width, and interface normal changes shading reference frame
+            // hence it also changes the directional relation between the visibility cone and the BSDF lobe.)
```
Comment (slunity) [inline com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/StackLitMasterNode.cs]: @sebastienlagarde  (just an obvious caveat)

File: com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLit.cs.hlsl [hunk 2]
```diff
@@ -57,9 +57,10 @@
 #define DEBUGVIEW_STACKLIT_SURFACEDATA_DIFFUSION_PROFILE_HASH (1133)
 #define DEBUGVIEW_STACKLIT_SURFACEDATA_SUBSURFACE_MASK (1134)
 #define DEBUGVIEW_STACKLIT_SURFACEDATA_THICKNESS (1135)
-#define DEBUGVIEW_STACKLIT_SURFACEDATA_SPECULAR_OCCLUSION_FIXUP_VISIBILITY_RATIO_THRESHOLD (1136)
-#define DEBUGVIEW_STACKLIT_SURFACEDATA_SPECULAR_OCCLUSION_FIXUP_STRENGTH (1137)
-#define DEBUGVIEW_STACKLIT_SURFACEDATA_SPECULAR_OCCLUSION_FIXUP_MAX_ADDED_ROUGHNESS (1138)
+#define DEBUGVIEW_STACKLIT_SURFACEDATA_SPECULAR_OCCLUSION_(CUSTOM_INPUT_ONLY) (1136)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLit.cs.hlsl]: caution, there is an error in the Macro here, you can't use parenthesis in the define [SurfaceDataAttributes("Specular Occlusion (custom input only)")]

File: com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLit.cs.hlsl [hunk 3]
```diff
@@ -309,6 +313,9 @@ void GetGeneratedSurfaceDataDebug(uint paramId, SurfaceData surfacedata, inout f
         case DEBUGVIEW_STACKLIT_SURFACEDATA_THICKNESS:
             result = surfacedata.thickness.xxx;
             break;
+        case DEBUGVIEW_STACKLIT_SURFACEDATA_SPECULAR_OCCLUSION_(CUSTOM_INPUT_ONLY):
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLit.cs.hlsl]: same here: caution, there is an error in the Macro here, you can't use parenthesis in the define [SurfaceDataAttributes("Specular Occlusion (custom input only)")]

**Comment by sebastienlagarde** [comment]
any katana link ?

**Comment by sebastienlagarde** [comment]
you can merge HDRP/staging and launch katana again, last katana was all red but due to bad state of master, this should be fixed now. thnaks

File: com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLit.cs.hlsl (hunk 2 above)
Comment (slunity) [reply com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLit.cs.hlsl]: Oops, obviously, thanks will fix now.

**Pull request merged**